beginner_source/blitz/tensor_tutorial.py

```diff
@@ -1,195 +1,189 @@
-# Construct a 5x3 matrix, uninitialized:
 
-x = torch.empty(5, 3)
-print(x)
- 
-###############################################################
-# Construct a randomly initialized matrix:
+######################################################################
+# Tensor Initialization
+# ~~~~~~~~~~~~~~~~~~~~~
+#
+# Tensors can be initialized in various ways. Take a look at the following examples
+#
+# **Directly from data / NumPy arrays:**
+# Tensors can be created directly by passing a Python list or sequence using the ``torch.tensor()`` constructor. The data type is automatically inferred from the data.
 
-x = torch.rand(5, 3)
-print(x)
+data = [[1, 2],[3, 4]]
+np_array = np.array(data)
 
-###############################################################
-# Construct a matrix filled zeros and of dtype long:
+tnsr_from_data = torch.tensor(data)
+tnsr_from_np = torch.from_numpy(np_array)
 
-x = torch.zeros(5, 3, dtype=torch.long)
-print(x)
 
 ###############################################################
-# Construct a tensor directly from data:
+# **From another tensor:**
+# The new tensor retains the properties of the arg tensor, unless explicitly overridden.
 
-x = torch.tensor([5.5, 3])
-print(x)
+new_ones_tnsr = torch.ones_like(tnsr_from_data) # 2 x 2 matrix of ones
 
-###############################################################
-# or create a tensor based on an existing tensor. These methods
-# will reuse properties of the input tensor, e.g. dtype, unless
-# new values are provided by user
+try:
+  new_rand_tnsr = torch.rand_like(tnsr_from_data) # 2 x 2 matrix of random numbers
+except RuntimeError as e:
+  print(f"RuntimeError thrown: {e}")
+  print()
+  print(f"Random values in PyTorch are floating points. Datatype passed to torch.rand_like(): {tnsr_from_data.dtype}")
 
-x = x.new_ones(5, 3, dtype=torch.double)      # new_* methods take in sizes
-print(x)
 
-x = torch.randn_like(x, dtype=torch.float)    # override dtype!
-print(x)                                      # result has the same size
+######################################################################
+# This works after we override the dtype property
+#
 
-###############################################################
-# Get its size:
+new_rand_tnsr = torch.rand_like(tnsr_from_data, dtype=torch.float)
+print(new_rand_tnsr)
 
-print(x.size())
 
-###############################################################
-# .. note::
-#     ``torch.Size`` is in fact a tuple, so it supports all tuple operations.
+######################################################################
+# **With random or constant values:**
 #
-# Operations
-# ^^^^^^^^^^
-# There are multiple syntaxes for operations. In the following
-# example, we will take a look at the addition operation.
-#
-# Addition: syntax 1
-y = torch.rand(5, 3)
-print(x + y)
 
-###############################################################
-# Addition: syntax 2
+shape = (2,3,)
```

Comment:
I think you should add a very brief explanation of shape here. It will be obvious to most but for some it a one sentence explanation will provide a lot of value.